R/aes.r

```diff
@@ -22,9 +22,10 @@ NULL
 #' programming vignette](http://dplyr.tidyverse.org/articles/programming.html)
 #' to learn more about these techniques.
 #'
-#' @param x,y,... List of name value pairs giving aesthetics to map to
-#'   variables. The names for x and y aesthetics are typically omitted because
-#'   they are so common; all other aesthetics must be named.
+#' @param x,y,... List of name-value pairs in the form `aesthetic = column_name`
+#'   describing which variables in the layer data should be mapped to which
+#'   aesthetics used by paired geom/stat. The names for x and y aesthetics are typically
+#'   omitted because they are so common; all other aesthetics must be named.
```

Comment:
I do think it's worth being explicit about the wrong way to do things here. Maybe start a new paragraph and say something like: It's not necessary to refer to the dataset in `aes()`. Instead of writing `aes(x = data$myvariable)`, all you need is `aes(myvariable)`.

Maybe @mine-cetinkaya-rundel has thoughts on this?

Reply:
I agree with @hadley that it's good to be explicit about what not to do, especially for those coming to ggplot2 from `plot()` where `data$column_name` usage is common. I might suggest matching the wording to the warning and say 

"It's discouraged to refer to the data in `aes()`. Instead of aes(x = data$variable), use aes(variable)." 

and similarly use `aesthetic = variable` earlier in the description (or use `column` or `column_name` or `variable_name` but keep it consistent throughout). I *think* it's more correct to avoid `_name` since it's not the name (character string) but the variable itself we're referring to, but I'm not sure.